### ISSUE TITLE
MPI_ERRORS_ARE_FATAL ?

### ISSUE THREAD

**Issue by jeff-regier** [tonyhffong/Lint.jl#149]
I'm getting the error below when I run `lintpkg("Celeste")`. Something about MPI? These Celeste package uses a library (Dtree.jl) that itself uses MPI, but it doesn't produce any errors on my system. I don't see why this would affect Lint. I'm using Julia 0.5-rc2 and master of Lint.jl.

```
julia> Pkg.add("Celeste")
INFO: No packages to install, update or remove
INFO: Package database updated

julia> Pkg.checkout("Celeste")
INFO: Checking out Celeste master...
INFO: Pulling Celeste latest master...
INFO: No packages to install, update or remove

julia> using Lint

julia> lintpkg("Celeste")
ParseError("extra token \"Log\" after end of expression")
import ..Log
ParseError("extra token \"Model\" after end of expression")
import ..Model.ParamSet
ParseError("extra token \"Model\" after end of expression")
import ..Model.getids
ParseError("extra token \"WCSUtils\" after end of expression")
import ..WCSUtils
ParseError("extra token \"SensitiveFloats\" after end of expression")
import ..SensitiveFloats.clear!
ParseError("extra token \"Log\" after end of expression")
import ..Log
ParseError("extra token \"SensitiveFloats\" after end of expression")
import ..SensitiveFloats.clear!
ParseError("extra token \"SensitiveFloats\" after end of expression")
import ..SensitiveFloats
ParseError("extra token \"Log\" after end of expression")
import ..Log
ParseError("extra token \"Model\" after end of expression")
import ..Model.RawPSF
ParseError("extra token \"Model\" after end of expression")
import ..Model.Image
ParseError("extra token \"Model\" after end of expression")
import ..Model.CatalogEntry
ParseError("extra token \"PSF\" after end of expression")
import ..PSF
ParseError("extra token \"Log\" after end of expression")
import ..Log
ParseError("extra token \"WCSUtils\" after end of expression")
import ..WCSUtils
ParseError("extra token \"PSF\" after end of expression")
import ..PSF
ParseError("extra token \"OptimizeElbo\" after end of expression")
import ..OptimizeElbo
ParseError("extra token \"Log\" after end of expression")
import ..Log
ParseError("invalid identifier name \".\"")
import .Log
ParseError("invalid identifier name \".\"")
import .SDSSIO
ParseError("invalid identifier name \".\"")
import .Infer
ParseError("invalid identifier name \".\"")
import .SDSSIO.RunCamcolField
[arina:23678] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_paffinity_hwloc: perhaps a missing symbol, or compiled for a different version of Open MPI? (ignored)
[arina:23678] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_carto_auto_detect: perhaps a missing symbol, or compiled for a different version of Open MPI? (ignored)
[arina:23678] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_carto_file: perhaps a missing symbol, or compiled for a different version of Open MPI? (ignored)
[arina:23678] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_shmem_mmap: perhaps a missing symbol, or compiled for a different version of Open MPI? (ignored)
[arina:23678] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_shmem_posix: perhaps a missing symbol, or compiled for a different version of Open MPI? (ignored)
[arina:23678] mca: base: component_find: unable to open /usr/lib/openmpi/lib/openmpi/mca_shmem_sysv: perhaps a missing symbol, or compiled for a different version of Open MPI? (ignored)
--------------------------------------------------------------------------
It looks like opal_init failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during opal_init; some of which are due to configuration or
environment problems.  This failure appears to be an internal failure;
here's some additional information (which may only be relevant to an
Open MPI developer):

  opal_shmem_base_select failed
  --> Returned value -1 instead of OPAL_SUCCESS
--------------------------------------------------------------------------
[arina:23678] [[INVALID],INVALID] ORTE_ERROR_LOG: Error in file runtime/orte_init.c at line 79
--------------------------------------------------------------------------
It looks like MPI_INIT failed for some reason; your parallel process is
likely to abort.  There are many reasons that a parallel process can
fail during MPI_INIT; some of which are due to configuration or environment
problems.  This failure appears to be an internal failure; here's some
additional information (which may only be relevant to an Open MPI
developer):

  ompi_mpi_init: orte_init failed
  --> Returned "Error" (-1) instead of "Success" (0)
--------------------------------------------------------------------------
*** An error occurred in MPI_Init
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL: your MPI job will now abort
[arina:23678] Local abort before MPI_INIT completed successfully; not able to aggregate error messages, and not able to guarantee that all other processes were killed!
```

**Comment by TotalVerb** [comment]
This is because Lint is not currently a fully static checker. While it won't evaluate any code in the module you're linting, it *will* import and execute all the modules that your module depends on. I consider this heavily undesirable behaviour, but it's not easy to fix.

**Comment by jeff-regier** [comment]
All the modules my module depends on load without errors though. Are they
getting loaded in a different way Lint.jl? Is there a workaround?

**Comment by TotalVerb** [comment]
@jeff-regier Unfortunately, the only workaround I am aware of is to comment out the `using Dtree` line and try to filter out all the "undefined symbol" warnings.

**Comment by TotalVerb** [comment]
This modification should probably be done in Lint, not in Dtree. I will take a look tonight.

**Comment by TotalVerb** [comment]
Sorry for the delay. See #153 for a fix.

**Comment by jeff-regier** [comment]
Looks like the tests are failing. Thanks for working on this.

**Comment by TotalVerb** [comment]
Test failures are unrelated.

**Comment by jeff-regier** [comment]
That branch works really well for me, totally solves the problem. Thanks much!
